bench/src/ffi/tcl.rs

```diff
@@ -134,13 +134,10 @@ impl<'r, 't> Iterator for FindMatches<'r, 't> {
     type Item = (usize, usize);
 
     fn next(&mut self) -> Option<(usize, usize)> {
-        match self.re.find_at(self.text, self.last_match) {
-            None => None,
-            Some((s, e)) => {
-                self.last_match = e;
-                Some((s, e))
-            }
-        }
+        self.re.find_at(self.text, self.last_match).map(|(s, e)| {
+            self.last_match = e;
+            (s, e)
+        })
```

Comment:
I personally am not a fan of using `map` combined with mutations on other objects. Maybe revert this change and others like it above?